api/routes/auth-routes.js

```diff
@@ -0,0 +1,64 @@
+const router = require("express").Router();
+const passport = require("passport");
+const CLIENT_HOME_PAGE_URL = "http://localhost:3000/homepage";
+
+// when login is successful, retrieve user info
+router.get("/login/success", (req, res) => {
+  if (req.user) {
+    res.json({
+      isAuthenticated: true,
+      message: "user has successfully authenticated",
+      user: req.user,
+      cookies: req.cookies
+    });
+  }
+});
+
+const authCheck = (req, res, next) => {
+  if (!req.user) {
+    res.status(401).json({
+      isAuthenticated: false,
+      message: "user has not been authenticated"
+    });
+  } else {
+    next();
+  }
+};
+
+router.get("/checkStatus", authCheck, (req, res) => {
+    passport.authenticate('google', { scope: ["profile"] }),
```

Comment:
I think at this point we don't need `passport.authenticate()` anymore

Reply:
checkStatus isn't quite working properly. I dont know if its because the authentication wasn't completed or because theres a bug with route itself